### ISSUE TITLE
Add support to reverse encryption

### ISSUE THREAD

**Issue by neurodroid** [neurodroid/cryptonite#39]
_From [salatiel...@gmail.com](https://code.google.com/u/103307377090736138145/) on August 24, 2012 11:18:47_

Please add support to use reverse encryption in criptonite.
In encfs reverse mode (encfs --reverse) you can have an unencrypted local folder  and a mount point that encrypts the files on the fly. So  we can make backups to dropbox encrypted while keeping the local data unencrypted removing any  overhead to access the files locally.

Example:

/sdcard/DCIM -> unencrypted
encfs --reverse /sdcard/DCIM /sdcard/dropbox_sync_pics 
/sdcard/dropbox_sync _pics  -> on the fly encrypted that will be synced with dropbox using third part apps.
So i can work locally with no overhead at all and only during sync my files will be transparently encrypted on dropbox servers.

Thanks !

_Original issue: http://code.google.com/p/cryptonite/issues/detail?id=39_

**Comment by dimovnike** [comment]
I would like this too. Please, make an option to start reverse encryption at boot with a saved obfuscated password.

**Comment by pepa65** [comment]
This is a prime use of encfs, and it's built-in! Such an easy solution in a cloud-storage world!! Want this for sure.

**Comment by dimovnike** [comment]
the problem is that the --reverse option itself works good from cmdline, the encfs can be started with a script from cmdline via some autostart app but the mounted dir is not seen by apps.... now if someone can find a good solution to this :)

**Comment by neurodroid** [comment]
@dimovnike I think it's an SELinux issue, as pointed out in issue #47 and issue #83. If someone could send a pull request addressing these issues (@xmikos seems to ave some experience) I'd be grateful.

**Comment by dimovnike** [comment]
Im more inclined to think its mount namespace issue, I am researching this, will get back when I have some results.

**Comment by dimovnike** [comment]
The new SuperSU has the option to join the master mount namespace. So while having last SuperSU the following command works:

su --mount-master -d -c "/data/data/csh.cryptonite/encfs --public --reverse --extpass='echo -n scrambled==|base64 -d' /storage/sdcard0/DCIM /storage/sdcard0/path/path/"

I got "Transport endpoint is not connected" a couple of times (i suspect the encfs dies, but why?). I can see the encrypted files from any app.

**Comment by neurodroid** [comment]
If we can find out how SuperSU achieves this, maybe we can port that functionality to Cryptonite?

**Comment by dimovnike** [comment]
yeah, im sure there is a way, but the problem now is how to avoid "Transport endpoint is not connected" errors. Do you think android kills encfs because its not run as as service?
